intake_stac/catalog.py

```diff
@@ -241,111 +306,106 @@ def _get_metadata(self, **kwargs):
         """
-        band_info = []
-        try:
-            # NOTE: ensure we test these scenarios
-            # FileNotFoundError: [Errno 2] No such file or directory: '/catalog.json'
+        # NOTE: maybe return entire dataframe w/ central_wavelength etc?
+        common2band = {}
+        # 1. try to get directly from item metadata
+        if 'eo' in self._stac_obj.stac_extensions:
+            eo = self._stac_obj.ext['eo']
+            for band in eo.bands:
+                common2band[band.common_name] = band.name
+
+        # 2. go a level up to collection metadata
+        if common2band == {}:
             collection = self._stac_obj.get_collection()
-            if 'item-assets' in collection.stac_extensions:
-                for val in collection.ext['item_assets']:
-                    if 'eo:bands' in val:
-                        band_info.append(val.get('eo:bands')[0])
-            else:
-                band_info = collection.summaries['eo:bands']
+            # Can simplify after item-assets extension implemented in Pystac
+            # https://github.com/stac-utils/pystac/issues/132
+            for asset, meta in collection.extra_fields['item_assets'].items():
+                eo = meta.get('eo:bands')
+                if eo:
+                    for entry in eo:
+                        common_name = entry.get('common_name')
+                        if common_name:
+                            common2band[common_name] = asset
 
-        except Exception:
-            for band in self._stac_obj.ext['eo'].get_bands():
-                band_info.append(band.to_dict())
-        finally:
-            if not band_info:
-                raise ValueError(
-                    'Unable to parse "eo:bands" information from STAC Collection or Item Assets'
-                )
-        return band_info
+        return common2band
 
-    def stack_bands(self, bands, path_as_pattern=None, concat_dim='band'):
+    def stack_assets(self, assets, path_as_pattern=None, concat_dim='band', override_coords=None):
         """
-        Stack the listed bands over the ``band`` dimension.
+        Stack the listed assets over the ``band`` dimension.
 
-        This method only works for STAC Items using the 'eo' Extension
-        https://github.com/radiantearth/stac-spec/tree/master/extensions/eo
-
-        NOTE: This method is not aware of geotransform information. It *assumes*
-        bands for a given STAC Item have the same coordinate reference system (CRS).
+        WARNING: This method is not aware of geotransform information. It *assumes*
+        assets for a given STAC Item have the same coordinate reference system (CRS).
         This is usually the case for a given multi-band satellite acquisition.
-        Coordinate alignment is performed automatically upon calling the
-        `to_dask()` method to load into an Xarray DataArray if bands have diffent
-        ground sample distance (gsd) or array shapes.
```

Comment:
i learned the hard way that this isn't exactly true. it works for landsat bands that were aligned based on GSD, but here are two with the same CRS and transform, different GSD that xarray fails to do what we expect... i'll open an issue to illustrate this (https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/M/YB/2020/8/S2A_36MYB_20200814_0_L2A/B04.tif, https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/M/YB/2020/8/S2A_36MYB_20200814_0_L2A/B05.tif)

Reply:
Hi @scottyhq,
i stumbled upon this, do you have a suggestion how to go about stacking (sen2) bands with different GSD to be later used for lazy-loading in a dask array?
Thanks,
Vitus

Reply:
@vitusbenson i'd recommend explicitly interpolating, or using a gdal vrt
```
url4 = 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/M/YB/2020/8/S2A_36MYB_20200814_0_L2A/B04.tif'
url5 = 'https://sentinel-cogs.s3.us-west-2.amazonaws.com/sentinel-s2-l2a-cogs/36/M/YB/2020/8/S2A_36MYB_20200814_0_L2A/B05.tif'
da4 = rioxarray.open_rasterio(url4, chunks='auto').assign_coords(band=['B4']) #gsd=10
da5 = rioxarray.open_rasterio(url5, chunks='auto').assign_coords(band=['B5']) #gsd=20
da4_interp = da4.interp(y=da5['y'], x=da5['x'],  method='nearest')
da = xr.concat([da4_interp, da5], dim='band')
da
```

Reply:
Thanks - I'll have to give that a shot to compare.

Reply:
Thank you! I wasn't aware of rioxarray's interpolation capabilities - really useful!